### PR TITLE
initial support for Evanell EZ200

### PR DESCRIPTION
**Pull request by tux2000**
New device support.

Tested locally but based on the codebased that is installed using `npm ci`

Issue for new device support: https://github.com/Koenkk/zigbee2mqtt/issues/10636

### CODE REVIEW

File: converters/fromZigbee.js [hunk 1]
```diff
@@ -4136,6 +4136,43 @@ const converters = {
...
+            switch (dp) {
+            case tuya.dataPoints.evanellChildLock:
+                return {child_lock: value ? 'LOCK' : 'UNLOCK'};
+            case tuya.dataPoints.evanellBattery:
+                return {battery: value};
+            case tuya.dataPoints.evanellHeatingSetpoint:
+                // value is already converted to a number in JS, and we deduced that it needs to be divided by 10
+                return {current_heating_setpoint: (value / 10).toFixed(1)};
+            case tuya.dataPoints.evanellLocalTemp:
+                return {local_temperature: parseFloat((value / 10).toFixed(1))};
```
Comment (Koenkk) [inline converters/fromZigbee.js:4155]: Don't use to fixed here and 2 lines above, it will convert the number to  a string

**Comment by tux2000** [comment]
I removed the `toFixed()` statements and also tried to run the current code by installing from the branch with `npm install tux2000/zigbee-herdsman-converters#evanell`. For some reason both sending and receiving data does not work any longer, see log below. Any suggestions what this could be caused by? Could this be due to some upstream changes that need to be accounted for in my code or did I miss to copy some changes when adapting the code in the repository?

<details>
<summary>Log</summary>

```
Zigbee2MQTT:error 2022-01-08 00:06:57: Publish 'set' 'system_mode' to '0x9035eafffe34b766' failed: 'Error: Command 0x9035eafffe34b766/1 manuSpecificTuya.dataRequest({"seq":0,"dpValues":[{"dp":2,"datatype":4,"data":[2]}]}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Parameter 'dp' is missing)'
Zigbee2MQTT:debug 2022-01-08 00:06:57: Error: Command 0x9035eafffe34b766/1 manuSpecificTuya.dataRequest({"seq":0,"dpValues":[{"dp":2,"datatype":4,"data":[2]}]}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Parameter 'dp' is missing)
    at ZclFrame.writePayloadCluster (/files/smarthome/zigbee/z2m3/zigbee2mqtt/node_modules/zigbee-herdsman/src/zcl/zclFrame.ts:154:23)
    at ZclFrame.toBuffer (/files/smarthome/zigbee/z2m3/zigbee2mqtt/node_modules/zigbee-herdsman/src/zcl/zclFrame.ts:78:18)
    at ZStackAdapter.sendZclFrameToEndpointInternal (/files/smarthome/zigbee/z2m3/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:388:22)
    at Object.func (/files/smarthome/zigbee/z2m3/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:356:25)
    at Queue.executeNext (/files/smarthome/zigbee/z2m3/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/queue.ts:32:42)
    at /files/smarthome/zigbee/z2m3/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/queue.ts:21:18
    at new Promise (<anonymous>)
    at Queue.execute (/files/smarthome/zigbee/z2m3/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/queue.ts:19:16)
    at ZStackAdapter.sendZclFrameToEndpoint (/files/smarthome/zigbee/z2m3/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:354:27)
    at /files/smarthome/zigbee/z2m3/zigbee2mqtt/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:723:39
Zigbee2MQTT:debug 2022-01-08 00:07:22: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,250],"type":"Buffer"},"datatype":2,"dp":62,"fn":0,"seq":8960}' from endpoint 1 with groupID 0
Zigbee2MQTT:warn  2022-01-08 00:07:22: zigbee-herdsman-converters:EvanellThermostat: NOT RECOGNIZED DP #62 with data {"seq":8960,"dp":62,"datatype":2,"fn":0,"data":{"type":"Buffer","data":[0,0,0,250]}}
Zigbee2MQTT:info  2022-01-08 00:07:22: Configuring '0x9035eafffe34b766'
Zigbee2MQTT:info  2022-01-08 00:07:22: Successfully configured '0x9035eafffe34b766'
Zigbee2MQTT:debug 2022-01-08 00:07:23: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,250],"type":"Buffer"},"datatype":2,"dp":4,"fn":0,"seq":9216}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:07:23: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"battery":null,"child_lock":null,"current_heating_setpoint":null,"linkquality":126}'
Zigbee2MQTT:debug 2022-01-08 00:07:23: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,211],"type":"Buffer"},"datatype":2,"dp":5,"fn":0,"seq":9472}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:07:23: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"battery":null,"child_lock":null,"current_heating_setpoint":null,"linkquality":126,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:07:23: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,38],"type":"Buffer"},"datatype":2,"dp":6,"fn":0,"seq":9728}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:07:23: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":126,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:07:23: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,211],"type":"Buffer"},"datatype":2,"dp":5,"fn":0,"seq":9984}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:07:23: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":126,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:07:24: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,38],"type":"Buffer"},"datatype":2,"dp":6,"fn":0,"seq":10240}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:07:24: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":126,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:08:19: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,240],"type":"Buffer"},"datatype":2,"dp":62,"fn":0,"seq":10496}' from endpoint 1 with groupID 0
Zigbee2MQTT:warn  2022-01-08 00:08:19: zigbee-herdsman-converters:EvanellThermostat: NOT RECOGNIZED DP #62 with data {"seq":10496,"dp":62,"datatype":2,"fn":0,"data":{"type":"Buffer","data":[0,0,0,240]}}
Zigbee2MQTT:debug 2022-01-08 00:08:19: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,240],"type":"Buffer"},"datatype":2,"dp":4,"fn":0,"seq":10752}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:08:20: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":63,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:08:20: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,211],"type":"Buffer"},"datatype":2,"dp":5,"fn":0,"seq":11008}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:08:20: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":63,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:08:20: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,45],"type":"Buffer"},"datatype":2,"dp":6,"fn":0,"seq":11264}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:08:20: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":63,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:08:20: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,211],"type":"Buffer"},"datatype":2,"dp":5,"fn":0,"seq":11520}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:08:20: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":63,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:08:20: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,45],"type":"Buffer"},"datatype":2,"dp":6,"fn":0,"seq":11776}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:08:20: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":63,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:09:23: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[7,0,210,8,0,150,18,0,220,23,0,150,7,0,210,8,0,150,18,0,220,23,0,220,7,0,210,8,0,250,18,0,220,23,0,150],"type":"Buffer"},"datatype":0,"dp":101,"fn":0,"seq":12032}' from endpoint 1 with groupID 0
Zigbee2MQTT:warn  2022-01-08 00:09:23: zigbee-herdsman-converters:EvanellThermostat: NOT RECOGNIZED DP #101 with data {"seq":12032,"dp":101,"datatype":0,"fn":0,"data":{"type":"Buffer","data":[7,0,210,8,0,150,18,0,220,23,0,150,7,0,210,8,0,150,18,0,220,23,0,220,7,0,210,8,0,250,18,0,220,23,0,150]}}
Zigbee2MQTT:debug 2022-01-08 00:09:24: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,250],"type":"Buffer"},"datatype":2,"dp":4,"fn":0,"seq":12288}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:09:24: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":78,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:09:24: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,250],"type":"Buffer"},"datatype":2,"dp":4,"fn":0,"seq":12544}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:09:24: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":76,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:09:24: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0],"type":"Buffer"},"datatype":4,"dp":2,"fn":0,"seq":12800}' from endpoint 1 with groupID 0
Zigbee2MQTT:warn  2022-01-08 00:09:24: zigbee-herdsman-converters:evanell_thermostat: Mode undefined is not recognized.
Zigbee2MQTT:debug 2022-01-08 00:09:24: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,250],"type":"Buffer"},"datatype":2,"dp":4,"fn":0,"seq":13056}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:09:24: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":78,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:09:25: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,212],"type":"Buffer"},"datatype":2,"dp":5,"fn":0,"seq":13312}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:09:25: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":81,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:09:25: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,48],"type":"Buffer"},"datatype":2,"dp":6,"fn":0,"seq":13568}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:09:25: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":84,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:09:25: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,212],"type":"Buffer"},"datatype":2,"dp":5,"fn":0,"seq":13824}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:09:25: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":94,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:09:25: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,48],"type":"Buffer"},"datatype":2,"dp":6,"fn":0,"seq":14080}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:09:25: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":94,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:09:26: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,212],"type":"Buffer"},"datatype":2,"dp":5,"fn":0,"seq":14336}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:09:26: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":68,"local_temperature":null}'
Zigbee2MQTT:debug 2022-01-08 00:09:26: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,48],"type":"Buffer"},"datatype":2,"dp":6,"fn":0,"seq":14592}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2022-01-08 00:09:26: MQTT publish: topic 'zigbee2mqtt/0x9035eafffe34b766', payload '{"child_lock":null,"current_heating_setpoint":null,"linkquality":86,"local_temperature":null}'
```
</details>

**Comment by Koenkk** [comment]
Ah yes, the dp value format is now a list, example changeset: https://github.com/Koenkk/zigbee-herdsman-converters/commit/0646fb4097fa8718ca9cdbaba0950d1f20fce57a

**Comment by tux2000** [comment]
I tried to adapt the code accordingly in fromZigbee, but its not working :( 

`Exception while calling fromZigbee converter: msg.data.dpValues is not iterable}`

<details>
<summary>Log</summary>

```
Zigbee2MQTT:debug 2022-01-08 11:23:50: {"seq":30976,"dp":4,"datatype":2,"fn":0,"data":{"type":"Buffer","data":[0,0,0,245]}}
Zigbee2MQTT:error 2022-01-08 11:23:50: Exception while calling fromZigbee converter: msg.data.dpValues is not iterable}
Zigbee2MQTT:debug 2022-01-08 11:23:51: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[7,0,210,8,0,150,18,0,220,23,0,150,7,0,210,8,0,150,18,0,220,23,0,220,7,0,210,8,0,250,18,0,235,23,0,245],"type":"Buffer"},"datatype":0,"dp":101,"fn":0,"seq":31232}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2022-01-08 11:23:51: {"seq":31232,"dp":101,"datatype":0,"fn":0,"data":{"type":"Buffer","data":[7,0,210,8,0,150,18,0,220,23,0,150,7,0,210,8,0,150,18,0,220,23,0,220,7,0,210,8,0,250,18,0,235,23,0,245]}}
Zigbee2MQTT:error 2022-01-08 11:23:51: Exception while calling fromZigbee converter: msg.data.dpValues is not iterable}
Zigbee2MQTT:debug 2022-01-08 11:23:51: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,223],"type":"Buffer"},"datatype":2,"dp":5,"fn":0,"seq":31488}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2022-01-08 11:23:51: {"seq":31488,"dp":5,"datatype":2,"fn":0,"data":{"type":"Buffer","data":[0,0,0,223]}}
Zigbee2MQTT:error 2022-01-08 11:23:51: Exception while calling fromZigbee converter: msg.data.dpValues is not iterable}
Zigbee2MQTT:debug 2022-01-08 11:23:51: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,39],"type":"Buffer"},"datatype":2,"dp":6,"fn":0,"seq":31744}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2022-01-08 11:23:51: {"seq":31744,"dp":6,"datatype":2,"fn":0,"data":{"type":"Buffer","data":[0,0,0,39]}}
Zigbee2MQTT:error 2022-01-08 11:23:51: Exception while calling fromZigbee converter: msg.data.dpValues is not iterable}
Zigbee2MQTT:debug 2022-01-08 11:23:51: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,223],"type":"Buffer"},"datatype":2,"dp":5,"fn":0,"seq":32000}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2022-01-08 11:23:51: {"seq":32000,"dp":5,"datatype":2,"fn":0,"data":{"type":"Buffer","data":[0,0,0,223]}}
Zigbee2MQTT:error 2022-01-08 11:23:51: Exception while calling fromZigbee converter: msg.data.dpValues is not iterable}
Zigbee2MQTT:debug 2022-01-08 11:23:51: Received Zigbee message from '0x9035eafffe34b766', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"data":{"data":[0,0,0,39],"type":"Buffer"},"datatype":2,"dp":6,"fn":0,"seq":32256}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2022-01-08 11:23:51: {"seq":32256,"dp":6,"datatype":2,"fn":0,"data":{"type":"Buffer","data":[0,0,0,39]}}
Zigbee2MQTT:error 2022-01-08 11:23:51: Exception while calling fromZigbee converter: msg.data.dpValues is not iterable}
```
</details>

I added a temporary debug message with the content of `msg.data` and for some reason I dont get the new dpValues format 😕

**Comment by tux2000** [comment]
Got a bit further with using the `dev` branch from https://github.com/Koenkk/zigbee2mqtt/tree/dev 🤦
Will get back when the code is working

**Comment by tux2000** [comment]
Now working locally

**Comment by tux2000** [comment]
This was my procedure to test locally:

```
mkdir new_folder # New folder
cd new_folder

git clone https://github.com/Koenkk/zigbee2mqtt.git # Clone and switch to dev
cd zigbee2mqtt
git checkout HEAD -- npm-shrinkwrap.json
git fetch
git checkout dev # Change 'dev' to 'master' to switch back to the release version
git pull

npm ci # install dependencies

npm install tux2000/zigbee-herdsman-converters#evanell # install evanell converters

# Edit configuration in data folder

npm start # Start the zigbee2mqtt deamon

# Pair the device
```

**Comment by Koenkk** [comment]
looks good, thanks!

**Pull request merged**